### DISCUSSION TITLE
How to deal with delay?

### DISCUSSION THREAD

**Discussion by bluescoobie1969** [ScratMan/HASmartThermostat#59] [Q&A]
In a room temperature control, there always will be a (significant) delay between the radiator starting to heat and when your temperature sensor begins to react. This delay can cause instability in the control loop, but certainly will cause issues for nice behaviour of the control loop. So how to deal with this delay? Any tips are welcome! Btw I'm planning to set up a system with a Eurotronic Spirit Zigbee TRV which let's me control valve position directly, together with  a Sonoff temperature sensor somewhere in the room.

**Comment by ScratMan** [comment]
Hello, that's the purpose of the PID : deal with regulation system delay and behaviour. There is nothing more to do than adjusting the kp, ki and kd gains (and optionally ke gain for outside temperature compensation with 2021.12.5 which is now in beta stage).

**Reply by bluescoobie1969** [reply]
Hi! Well....Delay is a control loop or Dead Time is very nasty to control. Especially if the delay is very long. Compare it with a car which wheels react only 10 sec later then turning the wheel. I remember from university long time ago, that delay in a Control Loop can cause instability. So you can only use small Kp and no Ki which will build up an enormous error-signal during the delay. So maybe you can compensate for the delay, but you will get a very bad control of the room temperature, slow, much overshoot. This is exactly what the Chinese TRV's suffer from. Because they made their control loop veeery slow and veeery sloppy. Only way to compensate for the delay. I remember that the Good Old Honeywel wall thermostate CT87 uses an anticipation element. A variable resistor in series which heats up when the termostat switches on. When adjusted well to the delay of the room, this improves the control behaviour dramatically!

**Reply by ScratMan** [reply]
This is the purpose of the derivative component of the PID : wait for the temperature increase to decrease the power, and even stop the heating much before reaching the target temperature, so that there is no or at least small overshoot.

**Reply by HuismAndre** [reply]
The following might be of some interest in this case: https://www.automation.com/en-us/articles/2003-1/how-to-control-a-process-with-long-dead-time
